packages/x-components/src/x-modules/semantic-queries/store/types.ts

```diff
@@ -4,13 +4,14 @@ import { XActionContext } from '../../../store/actions.types';
 import { XStoreModule } from '../../../store/store.types';
 import { SemanticQueriesConfig } from '../config.types';
 import { ConfigMutations } from '../../../store/utils/config-store.utils';
+import { QueryMutations, QueryState } from '../../../store/utils/query.utils';
 
 /**
  * SemanticQueries store state.
  *
  * @public
  */
-export interface SemanticQueriesState {
+export interface SemanticQueriesState extends QueryState {
```

Comment:
Aside from extending `QueryState` you have to remove the `query` type definition in the state. `QueryState` already has the `query` type definition in it, so you're basically duplicating the type definition.

There are several modules that do this wrong so remember to fix those too.

Reply:
Hi @albertjcuac, @CachedaCodes : I think this should be also done for mutations if we are extending `QueryMutations`: is that right?
Example below: 
![Captura de pantalla 2023-10-18 a les 8 47 59](https://github.com/empathyco/x/assets/21217131/04367204-7e34-4d36-96b4-c4efc128df51)

Reply:
Yes, that's true